### PR TITLE
Always add stack/FPU stack pointer to uses of hell node

### PR DESCRIPTION
**Pull request by ptomin**
`IndirectCallRewriter` works incorrectly if there is not stack pointers
in uses of indirect call.
This is the fix for #789

### CODE REVIEW

File: src/Decompiler/Analysis/SsaTransform.cs [hunk 1]
```diff
@@ -724,7 +724,11 @@ private void GenerateUseDefsForUnknownCallee(CallInstruction ci)
                     .Select(s => s.Instruction)
                     .OfType<DefInstruction>()
                     .Select(d => d.Identifier)
-                    .Where(i => ssa.Identifiers[i].Uses.Count > 0));
+                    .Where(i => ssa.Identifiers[i].Uses.Count > 0))
+                .Concat(
```
[the remark this answers is not in the document]
Comment (ptomin) [reply src/Decompiler/Analysis/SsaTransform.cs]: Done in 1a8a3ca. Is it enough?

File: subjects/Elf/RiscV/ipcalc/ipcalc.c [hunk 2]
```diff
@@ -2048,9 +2051,13 @@ struct Eq_n * fn0000000000017B40(word64 ra, struct Eq_n * gp, Eq_n a0, ptr64 a1,
 			word64 fa5_n;
 			word64 t1_n;
 			word64 fa3_n;
+			word64 sp_n;
 			a5_n();
 			if (a0_n != null)
+			{
 				*s3_n = (union Eq_n *) __strdup(a0_n);
+				sp_n += ~0x07;
```
[the remark this answers is not in the document]
Comment (ptomin) [reply subjects/Elf/RiscV/ipcalc/ipcalc.c]: Maybe unresolved hell nodes (like `a5_n()`) hold this `sp_n` identifier. I'll do more detailed investigation during this weekend. I think it should not prevent merge this PR. Appearing stack pointer assignments is still better than regression in `IncorrectUserSignature` sample

[the remark this answers is not in the document]
Comment (ptomin) [reply subjects/Elf/RiscV/ipcalc/ipcalc.c]: Strange. Stack delta for `strdup` is `0` on `Riscv`, return address on stack is `8`, so `Reko` scanner adjust stack after call which should not be done. Return address on stack should be changed to `0` or stack delta should be changed to `8`

**Comment by uxmal** [comment]
OK, let's grab the PR and continue investigating the spurious stack pointer references.

File: subjects/Elf/RiscV/ipcalc/ipcalc.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ptomin) [reply subjects/Elf/RiscV/ipcalc/ipcalc.c]: See #794 for possible fix

**Pull request merged**